### DISCUSSION TITLE
IPED-SearchApp.exe performance problem (8 GB ram)

### DISCUSSION THREAD

**Discussion by gisms** [sepinf-inc/IPED#1770] [Q&A] [answered]
Hi everyone;

We have a considerable size ufdr file, 300gb, indexed by iped.
When opening on machines with 8gb of ram, it closes abruptly, generating an error log:

_There is insufficient memory for the Java Runtime Environment to continue.
Native memory allocation (mmap) failed to map 36700160 bytes for Failed to commit area from 0x00000258cf450000 to 0x00000258d1750000 of length 36700160.
-------------- S U M M A R Y ------------
Command Line: -Xms1G -Xmx30G -Xms5G ...._

We start to open ipedsearch via bat with the following memory delimitation:

Command Line: -Xms1G -Xmx4G -Xms4G.

- For this machine scenario with 8gb of ram, would it be possible to improve the parameters?
- Another doubt, is there a way to change the command line without having to create a bat file? so that IPED-SearchApp.exe remains the default opening for the user

**Comment by patrickdalla** [comment]
do you use the latest 4.1.3 version?

Em sex., 14 de jul. de 2023 07:28, gisms ***@***.***>
escreveu:

**Reply by patrickdalla** [reply, marked as answer]
Try to edit resultsetviewers.xml in conf folder of the case, and comment the timeline viewer entry.
This disables timeline that consumes a lot of memory at the start. Maybe that is the source.

If that solves the startup, try to click on timelineview toogle button (clock icon).

If that is the problem, maybe you can continue the analisys without this.

Anyway, we must find some solution wnd this case could help us. Do you already use the 4.1.3 version?

The case we saw a similar problem there was tons of useless (at least for the case) activitysensordata entries with timestamps (>90% of timestamp info of the case) . Maybe you can generate the UFDR without this too.

**Reply by gisms** [reply]
In fact the problem stopped when removing the timeline. thank you very much.

Anyway, the videos files are showing as invalid reference when trying to open, so I'll leave it indexing in version 4.1.3.

I will evaluate the behavior of the timeline in the new version. I will bring a feedback about it.

**Reply by lfcnassif** [reply]
Not sure what you meant by invalid reference, but if UI asks for original evidence path, probably it was moved to another folder or not sent together with the case.

**Comment by gisms** [comment]
No.

An older case, It was 4.1.1

**Reply by patrickdalla** [reply]
"Command Line: -Xms1G -Xmx30G -Xms5G ...." this entry is from the log right?

If so, the JVM is actually starting with -Xmx30G. We found out that JVM really believes in this information, althought the machine have only 8GB, causing (guess) some garbage collection postpone, as it believes there is more memory, and this crash occurs.

We discussed this in https://github.com/sepinf-inc/IPED/issues/1712, and found a solution.

**Reply by gisms** [reply]
here we process the case on 64gb machines. but the analysis is done in cases of 8gb, maximum 16gb.

**Reply by lfcnassif** [reply]
You can also try to replace IPEDSearchApp.exe, iped/lib and iped/localization folders into the case for those from 4.1.3 version and test if it fixes the crash.

**Reply by lfcnassif** [reply]
But the Time chart still uses a reasonable amount of memory when the feature is used and it will be optimized probably in the future 4.2.0, without scheduled release date.
